create-docker-action.py

```diff
@@ -0,0 +1,78 @@
+import json
+import os
+import pathlib
+
+DESCRIPTION = 'description'
+REQUIRED = 'required'
+
+EVENT = os.environ['EVENT']
+REF = os.environ['REF']
+REPO = os.environ['REPO']
+REPO_OWNER_ID = os.environ['REPO_OWNER_ID']
+REPO_OWNER_ID_PYPA = '647025'
+
+
+def set_image(event: str, ref: str, repo: str, repo_owner_id: str) -> str:
+    if event == 'pull_request' and repo_owner_id == REPO_OWNER_ID_PYPA:
+        return '../../../Dockerfile'
+    docker_ref = ref.replace('/', '-')
+    return f'docker://ghcr.io/{repo}:{docker_ref}'
+
+
+image = set_image(EVENT, REF, REPO, REPO_OWNER_ID)
+
+action = {
+    'name': '🏃',
+    DESCRIPTION: (
+        'Run Docker container to upload Python distribution packages to PyPI'
+    ),
+    'inputs': {
+        'user': {DESCRIPTION: 'PyPI user', REQUIRED: False},
+        'password': {
+            DESCRIPTION: 'Password for your PyPI user or an access token',
+            REQUIRED: False,
+        },
+        'repository-url': {
+            DESCRIPTION: 'The repository URL to use',
+            REQUIRED: False,
+        },
+        'packages-dir': {
+            DESCRIPTION: 'The target directory for distribution',
+            REQUIRED: False,
+        },
+        'verify-metadata': {
+            DESCRIPTION: 'Check metadata before uploading',
+            REQUIRED: False,
+        },
+        'skip-existing': {
+            DESCRIPTION: (
+                'Do not fail if a Python package distribution'
+                ' exists in the target package index'
+            ),
+            REQUIRED: False,
+        },
+        'verbose': {DESCRIPTION: 'Show verbose output.', REQUIRED: False},
+        'print-hash': {
+            DESCRIPTION: 'Show hash values of files to be uploaded',
+            REQUIRED: False,
+        },
+    },
+    'runs': {
+        'using': 'docker',
+        'image': image,
+        'args': [
+            '${{ inputs.user }}',
+            '${{ inputs.password }}',
+            '${{ inputs.repository-url }}',
+            '${{ inputs.packages-dir }}',
+            '${{ inputs.verify-metadata }}',
+            '${{ inputs.skip-existing }}',
+            '${{ inputs.verbose }}',
+            '${{ inputs.print-hash }}',
+        ],
```

Comment:
I think nothing would change if we were to drop this. The positional arguments are not being used.
```suggestion
```

Reply:
Please see my previous comment (https://github.com/pypa/gh-action-pypi-publish/pull/230#discussion_r1616374020). Aren't the inputs used by the entrypoint script?

https://github.com/pypa/gh-action-pypi-publish/blob/699cd6103f50bf5c3b2f070c70712d109c168e6c/twine-upload.sh#L37-L41

Reply:
@br3ndonland I actually explained in the following comment https://github.com/pypa/gh-action-pypi-publish/pull/230#discussion_r1632408156 that those are read from environment variables and not from the positional CLI arguments:
https://github.com/pypa/gh-action-pypi-publish/blob/699cd6103f50bf5c3b2f070c70712d109c168e6c/twine-upload.sh#L21-L34

Reply:
If you look into the log, you'll see these as both env vars and pos args: https://github.com/pypa/gh-action-pypi-publish/actions/runs/9440557111/job/26000047753?pr=230#step:11:433.

Reply:
Ok right, thanks for explaining that. I accepted the suggestion.